File: src/scala/Average.scala
object Average extends App {
  val amounts = Seq(125.5, -25.0, 900.0, 42.0)
  val average = amounts.sum / amounts.length
  val negativeAmounts = amounts.count(_ < 0)
  println(s"average=$average negative_amounts=$negativeAmounts")
}

File: src/rust/average.rs
fn main() {
    let amounts = vec![125.5, -25.0, 900.0, 42.0];
    let sum: f64 = amounts.iter().sum();
    let negative_amounts = amounts.iter().filter(|x| **x < 0.0).count();
    println!("average={:.2} negative_amounts={}", sum / amounts.len() as f64, negative_amounts);
}
